packages/app-store/basecamp3/api/projects.ts

```diff
@@ -0,0 +1,72 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { defaultHandler, defaultResponder } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+
+import type { BasecampToken } from "../lib/CalendarService";
+import { userAgent } from "../lib/constants";
+import { refreshAccessToken } from "../lib/helpers";
+
+async function getHandler(req: NextApiRequest, res: NextApiResponse) {
+  const credential = await prisma.credential.findFirst({
+    where: {
+      userId: req.session?.user?.id,
+    },
+  });
+  if (!credential) {
+    return res.status(400).json({ message: "No app credential found for user" });
+  }
+  let credentialKey = credential.key as BasecampToken;
+  if (credentialKey.expires_at < Date.now()) {
+    credentialKey = (await refreshAccessToken(credential)) as BasecampToken;
+  }
+  const url = `${credentialKey.account.href}/projects.json`;
+
+  const resp = await fetch(url, {
+    headers: { "User-Agent": userAgent, Authorization: `Bearer ${credentialKey.access_token}` },
+  });
+  const data = await resp.json();
+  return res.json({ data, currentProject: credentialKey.projectId });
+}
+
+async function postHandler(req: NextApiRequest, res: NextApiResponse) {
+  const { projectId } = req.query;
+  const credential = await prisma.credential.findFirst({
+    where: {
+      userId: req.session?.user?.id,
+    },
+  });
+
+  if (!credential) {
+    return res.status(400).json({ message: "No app credential found for user" });
+  }
+  let credentialKey = credential.key as BasecampToken;
+
+  if (credentialKey.expires_at < Date.now()) {
+    credentialKey = (await refreshAccessToken(credential)) as BasecampToken;
+  }
+  // get schedule id
+  const basecampUserId = credentialKey.account.id;
+  const scheduleResponse = await fetch(
+    `https://3.basecampapi.com/${basecampUserId}/projects/${projectId}.json`,
```

Comment:
Why is this URL so different from the ones used to authenticate? is 37signals the company behind basecamp and they differentiate authenticating using their top company auth system and then use specific APIs from the product? Looks weird, that's all.

Reply:
Yep, 37signals for some reason uses launchpad.37signals.com to authorise users for all their applications it seems.